pkg/roachprod/vm/aws/aws.go

```diff
@@ -1315,21 +1405,33 @@ func (p *Provider) runInstance(
 	ebsVolumes := assignEBSVolumes(&opts, providerOpts)
 	args, err = genDeviceMapping(ebsVolumes, args)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	if providerOpts.UseSpot {
-		return runSpotInstance(l, p, args, az.Region.Name)
+		return runSpotInstance(l, p, args, az.Region.Name, providerOpts)
 		//todo(babusrithar): Add fallback to on-demand instances if spot instances are not available.
 	}
 	runInstancesOutput := RunInstancesOutput{}
-	return p.runJSONCommand(l, args, &runInstancesOutput)
+	err = p.runJSONCommand(l, args, &runInstancesOutput)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(runInstancesOutput.Instances) == 0 {
+		return nil, errors.Errorf("No instances found for run-instances command")
+	}
+
+	v := runInstancesOutput.Instances[0].toVM(map[string]vm.Volume{}, providerOpts.RemoteUserName)
```

Comment:
Without the volumes, `propagateDiskLabels` would be out of sync. We need the labels on all the volumes, primarily for cost-tracking.

Reply:
This actually doesn't differ with the current behavior:

- After creation, SetupSSH [Syncs](https://github.com/cockroachdb/cockroach/blob/master/pkg/roachprod/roachprod.go#L646) without the `IncludeVolumes` flag
- So [List](https://github.com/cockroachdb/cockroach/blob/master/pkg/roachprod/vm/aws/aws.go#L1131) doesn't know anything about the non-bootable volumes